package-system/AWSNativeSDK/README.md

```diff
@@ -0,0 +1,101 @@
+# Build New AWS Service Target in AWSNativeSDK
+First check package directories at [aws/aws-sdk-cpp](https://github.com/aws/aws-sdk-cpp) on GitHub. 
+The package name is the suffix of the directory name for the service.
+
+```
+aws-sdk-cpp\aws-cpp-sdk-<packageName>   # Repo directory name and packageName
+aws-sdk-cpp\aws-cpp-sdk-s3              # Example: Package name is s3
+```
+
+Note: If your target service is not available in current AWSNativeSDK version (but it is listed in aws-sdk-cpp repo),
+then please upgrade AWSNativeSDK version first.
+
+## Example: Build ElasticSearch (ES)
+
+### 1. Check [aws/aws-sdk-cpp](https://github.com/aws/aws-sdk-cpp) for aws-cpp-sdk-es
+
+### 2. Add ElasticSearch as part of AWSNativeSDK build target
+Modify AWSNativeSDK build script by adding ElasticSearch as a build target, use windows build script as a reference:
+
+[AWSNativeSDK build script (windows)](https://github.com/o3de/3p-package-source/blob/main/package-system/AWSNativeSDK/build_AWSNativeSDK_windows.cmd)
+```diff
+--- a/package-system/AWSNativeSDK/build_AWSNativeSDK_windows.cmd
++++ b/package-system/AWSNativeSDK/build_AWSNativeSDK_windows.cmd
+@@ -58,7 +58,7 @@ call cmake -S %SRC_PATH% -B %BLD_PATH%\%BUILD_TYPE%_%LIB_TYPE% ^
+            -DTARGET_ARCH=WINDOWS ^
+            -DCMAKE_CXX_STANDARD=17 ^
+            -DCPP_STANDARD=17 ^
+-           -DBUILD_ONLY="access-management;cognito-identity;cognito-idp;core;devicefarm;dynamodb;gamelift;identity-management;kinesis;lambda;mobileanalytics;queues;s3;sns;sqs;sts;transfer" ^
++           -DBUILD_ONLY="access-management;cognito-identity;cognito-idp;core;devicefarm;dynamodb;es;gamelift;identity-management;kinesis;lambda;mobileanalytics;queues;s3;sns;sqs;sts;transfer" ^
+            -DENABLE_TESTING=OFF ^
+            -DENABLE_RTTI=ON ^
+            -DCUSTOM_MEMORY_MANAGEMENT=ON ^
+```
+aws-cpp-sdk-es shared and static libraries will be generated after AWSNativeSDK build.
+
+### 3. Create a 3rdParty library target for ElasticSearch
+Modify FindAWSNativeSDK.cmake file by creating a new target for ElasticSearch, use windows cmake file as a reference:
+
+[FindAWSNativeSDK.cmake.Windows](https://github.com/o3de/3p-package-source/blob/main/package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.Windows)
+```diff
+--- a/package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.Windows
++++ b/package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.Windows
+@@ -272,6 +272,13 @@ ly_declare_aws_library(
+         aws-cpp-sdk-transfer
+ )
+
++#### ElasticSearch ####
++ly_declare_aws_library(
++    NAME
++        ElasticSearch
++    LIB_FILE
++        aws-cpp-sdk-es
++)
+```
+It is for consumer to pull aws-cpp-sdk-es shared and static libraries as build and runtime dependencies by using `3rdParty::AWSNativeSDK::ElasticSearch` target.
+
+### 4. Update AWSNativeSDK with correct version and revision
+Modify build config and list file to bump AWSNativeSDK with correct version and revision value, use windows build as a reference:
```

Comment:
Not sure if are covering the difference between providing a new version or providing a new revision well. Check that we have enough guidance about when we set the version vs when to set the revision

Reply:
This format is not for AWSNativeSDK package only, user should have enough guidance by onboarding this 3p system.